config/sampler_config.go

```diff
@@ -45,12 +47,379 @@ type RulesBasedSamplerCondition struct {
+					return n < conditionValue
+				}
+				return false
+			}
+			return nil
+		case "<=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToInt64(spanValue); exists && ok {
+					return n <= conditionValue
+				}
+				return false
+			}
+			return nil
+		}
+	case float64:
+		// check if conditionValue and spanValue are not equal
+		switch condition {
+		case "!=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n != conditionValue
+				}
+				return false
+			}
+			return nil
+		case "=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n == conditionValue
+				}
+				return false
+			}
+			return nil
+		case ">":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n > conditionValue
+				}
+				return false
+			}
+			return nil
+		case ">=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n >= conditionValue
+				}
+				return false
+			}
+			return nil
+		case "<":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n < conditionValue
+				}
+				return false
+			}
+			return nil
+		case "<=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n, ok := tryConvertToFloat(spanValue); exists && ok {
+					return n <= conditionValue
+				}
+				return false
+			}
+			return nil
+		}
+	case bool:
+		// check if conditionValue and spanValue are not equal
+		switch condition {
+		case "!=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n := tryConvertToBool(spanValue); exists && n {
+					return n != conditionValue
+				}
+				return false
+			}
+			return nil
+		case "=":
+			r.Matches = func(spanValue any, exists bool) bool {
+				if n := tryConvertToBool(spanValue); exists && n {
+					return n == conditionValue
+				}
+				return false
+			}
+			return nil
+		}
+	default:
+		return fmt.Errorf("%s value's type must be either string, int, float or bool, but was '%T'", conditionValue, conditionValue)
+	}
+	return nil // what return was missing?
+}
+
+func setMatchStringBasedOperators(r *RulesBasedSamplerCondition, condition string) error {
+	conditionValue, ok := r.Value.(string)
+	if !ok {
+		return fmt.Errorf("%s value must be a string, but was '%s'", condition, r.Value)
+	}
+
+	r.Matches = func(spanValue any, exists bool) bool {
+		s, ok := spanValue.(string)
+		if !ok {
+			return false
+		}
+
+		switch condition {
+		case "starts-with":
+			return strings.HasPrefix(s, conditionValue)
+		case "contains":
+			return strings.Contains(s, conditionValue)
+		case "does-not-contain":
+			return !strings.Contains(s, conditionValue)
+		}
+		return false
+	}
+	return nil
```

Comment:
nit: similar to the CompareOperators can we do this switch in `setMatchStringBasedOperators` and split each case into its own match function?

Reply:
Yeah, this makes it more consistent. I added this change in `setMatchStringBasedOperators`.